internal/handlers/handler_test.go

```diff
@@ -80,6 +80,90 @@ func handle(ctx context.Context, t *testing.T, handler *Handler, req *types.Docu
 	return actual
 }
 
+func TestFindArray(t *testing.T) {
+	t.Parallel()
+	ctx, handler, pool := setup(t, nil)
+	schema := testutil.Schema(ctx, t, pool)
+	loadArrayTestData(ctx, t, handler, schema)
```

Comment:
Let's start making a test dataset that would be shared between various query tests. I was thinking about documents with two fields: string `_id` for identification and `value` of various types. That would allow us to test edge cases like calling `$size` operators on NaN double value. https://github.com/FerretDB/dance/pull/33/files#diff-ee104391e4d88f69dce464c4f136bdeeb5c3d9a3b494a639416e6e0454afd4c2R86-R157

Reply:
I tried to mimic, that helped to catch new a bug.
As I got from documentation `$size` is used only with arrays.